Makefile

```diff
@@ -29,15 +33,28 @@ AFLAGS   = -mthumb -mcpu=cortex-m3
 LDFLAGS  = $(AFLAGS) -T$(LIB_PATH)/Device/SiliconLabs/EFM32GG/Source/GCC/efm32gg.ld
 LDFLAGS += -Wl,--start-group -lgcc -lc -lnosys -Wl,--end-group
 
+ifeq ($(PLATFORM),Darwin)
+LDFLAGS += target/thumbv7m-none-eabi/emlib-$(HASH).o
+LIB_DEPENENCIES = $(TARGET_DIR)/libemlib-$(HASH).rlib $(TARGET_DIR)/emlib-$(HASH).o
+else
+LIB_DEPENENCIES = $(TARGET_DIR)/libemlib-$(HASH).rlib
+endif
+
 RUSTFLAGS  = --target $(TARGET) --crate-type bin
 RUSTFLAGS += -g -C link-args="$(LDFLAGS)"
-RUSTFLAGS += -L $(LIB_DIR) -L $(TARGET_DIR) -L $(TARGET_DIR)/deps -L $(TARGET_DIR)/build/emlib-9b44da1ad5dde3c6/out
+RUSTFLAGS += -L $(LIB_DIR) -L $(TARGET_DIR) -L $(TARGET_DIR)/deps -L $(TARGET_DIR)/build/emlib-$(HASH)/out
 RUSTFLAGS += --emit=dep-info,link --verbose
 
 FLASHFLAGS = --verify --reset
 
-%.elf: $(PROJ_DIR)/$(PROJ_NAME).rs
+$(TARGET_DIR)/libemlib-$(HASH).rlib:
 	cargo build --target thumbv7m-none-eabi
+
```

Comment:
It's better to keep this as it was, and always trigger a "build" of libemlib. The Makefile don't know if the library has changed or not, cargo does.